src/Nethermind/Nethermind.Specs/ChainSpecStyle/ChainSpecBasedSpecProvider.cs

```diff
@@ -207,7 +207,7 @@ private static ReleaseSpec CreateReleaseSpec(ChainSpec chainSpec, long releaseSt
             releaseSpec.ValidateChainId = (chainSpec.Parameters.ValidateChainIdTransition ?? 0) <= releaseStartBlock;
             releaseSpec.ValidateReceipts = ((chainSpec.Parameters.ValidateReceiptsTransition > 0) ? Math.Max(chainSpec.Parameters.ValidateReceiptsTransition ?? 0, chainSpec.Parameters.Eip658Transition ?? 0) : 0) <= releaseStartBlock;
 
-            releaseSpec.Eip1559FeeCollector = releaseSpec.IsEip1559Enabled && (chainSpec.Parameters.Eip1559FeeCollectorTransition ?? long.MaxValue) <= releaseStartBlock ? chainSpec.Parameters.Eip1559FeeCollector : null;
+            releaseSpec.FeeCollector = releaseSpec.IsEip1559Enabled && (chainSpec.Parameters.Eip1559FeeCollectorTransition ?? long.MaxValue) <= releaseStartBlock ? chainSpec.Parameters.FeeCollector : null;
```

Comment:
should be also based on either or `Eip4844FeeCollectorTransition`

Reply:
I'm not sure we want to do this as the fee collector should still be enabled before EIP4844-Pectra is enabled

Reply:
yeah it should be one or the other

Reply:
Sorry I don't quite follow what you mean, we could have 3 possibilities: (a) no fee collection, (b) just EIP1559, (c) both EIP1559 and EIP4844. We set the fee collector in cases (b) and (c) so we only have to check that EIP1559 is enabled

Reply:
```suggestion
            releaseSpec.FeeCollector = (releaseSpec.IsEip1559Enabled && (chainSpec.Parameters.Eip1559FeeCollectorTransition ?? long.MaxValue) <= releaseStartBlock) || (releaseSpec.IsEip4844Enabled && (chainSpec.Parameters.Eip4844FeeCollectorTransition ?? long.MaxValue) <= releaseStartBlock))? chainSpec.Parameters.FeeCollector : null;
```

Reply:
I did this and also refactored [here](https://github.com/NethermindEth/nethermind/pull/6866/files#diff-b8ea3d5dd4a8bedda3ec0900ae86bf6c3352c28fcec738e3064511be8c9c52e7R601)  to make the 1559 and 4844 fee collection separately configurable